pkg/platform/kube/apigateway.go

```diff
@@ -0,0 +1,38 @@
+package kube
+
+import (
+	"github.com/nuclio/nuclio/pkg/platform"
+
+	"github.com/nuclio/errors"
+)
+
+func ValidateUpstreamSpec(apiGatewaySpec *platform.APIGatewaySpec) error {
+	upstreams := apiGatewaySpec.Upstreams
+
+	if len(upstreams) > 2 {
+		return errors.New("Received more than 2 upstreams. Currently not supported")
+	} else if len(upstreams) == 0 {
+		return errors.New("One or more upstreams must be provided in spec")
+	} else if apiGatewaySpec.Host == "" {
+		return errors.New("Host must be provided in spec")
+	}
+
+	// TODO: update this when adding more upstream kinds. for now allow only `nucliofunction` upstreams
+	kind := upstreams[0].Kind
+	if kind != platform.APIGatewayUpstreamKindNuclioFunction {
```

Comment:
For future compatibility / extensibility reasons - suggest this be a switch case with handling of the different upstream kinds (today just that)

Reply:
instead of using switch case, I chose to use a list of supported kinds, because it makes more sense as we only need to check if it is valid kind, and otherwise return an error